packages/flutter_hooks/lib/src/deferred_value.dart

```diff
@@ -0,0 +1,74 @@
+part of 'hooks.dart';
+
+/// Returns a [ValueNotifier] that updates its value with a timeout
+/// for the given input [toDefer] and triggers widget rebuilds accordingly.
+ValueNotifier<T> useDeferredValue<T>(
+  T toDefer, {
+  Duration timeout = const Duration(seconds: 1),
```

Comment:
Where is that "one second" comes from? That delay looks arbitrary to me

Reply:
maybe making `timeout` optional, if null or 0ms it could default to `300ms` and always combined with `WidgetsBinding.instance.addPostFrameCallback` then after that combination calling a setState to rebuild the widget and work with the deferred value ?

Reply:
We'd want a default similar to that of React, so definitely not 1 second.
If it's "right after the current render has completed", then we should do that too